deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+        Additionally, adjusts the number of slots in all tuples to the desired number: if there are fewer
+        slots than desired, will add empty lists at the end, and if there are more will
+        concatenate object slots.
+        Also, pads the number of answer options, the number of tuples per answer option, and the number
+        of background tuples to the specified lengths, adding empty ones as needed.
+        Note: Adjusting the number of slots happens before adjusting the lengths of all the slots.
+        since truncating is done from the left, the earlier object slots ar more likely to be
+        lost.
+        """
+        desired_num_options = max_lengths['num_options']
+        desired_num_question_tuples = max_lengths['num_question_tuples']
+        desired_num_background_tuples = max_lengths['num_background_tuples']
+        desired_num_slots = max_lengths['num_slots']
+        desired_question_slot_length = max_lengths['question_slot_length']
+        desired_background_slot_length = max_lengths['background_slot_length']
+        # Pad the number of answers
+        num_options = len(self.answers_indexed)
+        if num_options > desired_num_options:
+            raise Exception("Case of too many answer options ({0} provided) isn't currently "
+                            "handled.".format(num_options))
+        elif num_options < desired_num_options:
+            # Add empty answer option
+            additional_options = [[] for _ in range(desired_num_options - num_options)]
+            self.answers_indexed += additional_options
+        # Pad the number of tuples per option
+        for curr_answer in range(len(self.answers_indexed)):
+            self.answers_indexed[curr_answer] = self.pad_num_tuples(self.answers_indexed[curr_answer],
+                                                                    desired_num_question_tuples)
+        # Pad each of the tuples for slot length and words per slot
+        for answer_index in range(len(self.answers_indexed)):
+            self.answers_indexed[answer_index] = [self.pad_tuple(answer_tuple, desired_num_slots,
+                                                                 desired_question_slot_length)
+                                                  for answer_tuple in self.answers_indexed[answer_index]]
+        # Pad the number of background tuples
+        self.background_indexed = self.pad_num_tuples(self.background_indexed, desired_num_background_tuples)
+        # Pad each of the background tuples for number of slots and slot length
+        self.background_indexed = [self.pad_tuple(background_tuple, desired_num_slots,
+                                                  desired_background_slot_length)
+                                   for background_tuple in self.background_indexed]
+
+    # Helper method to pad a list of tuples to the desired number of tuples
+    @staticmethod
+    def pad_num_tuples(tuples_in, desired_length):
```

Comment:
Prefer using `self.pad_sequence_to_length` - it'll make this code shorter, re-use an already well-test piece of code, and be more flexible, because that method has more options than you have here.